Source/JavaScriptCore/runtime/StringPrototypeInlines.h

```diff
@@ -240,6 +240,61 @@ ALWAYS_INLINE JSString* stringReplaceStringString(JSGlobalObject* globalObject,
     return jsString(vm, WTFMove(result));
 }
 
+template<StringReplaceSubstitutions substitutions, StringReplaceUseTable useTable, typename TableType>
+ALWAYS_INLINE JSString* stringReplaceAllStringString(JSGlobalObject* globalObject, JSString* stringCell, const String& string, const String& search, const String& replacement, const TableType* table)
+{
+    VM& vm = globalObject->vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
+    Vector<size_t, 16> matchStarts;
+    size_t searchLength = search.length();
+
+    size_t matchStart = 0;
+    while (true) {
+        if constexpr (useTable == StringReplaceUseTable::Yes)
+            matchStart = table->find(StringView(string).substring(matchStart), search);
+        else
+            matchStart = StringView(string).find(vm.adaptiveStringSearcherTables(), StringView(search), matchStart);
+
+        if (matchStart == notFound)
+            break;
+
+        matchStarts.append(matchStart);
+        matchStart += searchLength;
+        if (!search.length())
+            ++matchStart;
+    }
+
+    if (matchStarts.isEmpty())
+        return stringCell;
+
+    size_t resultLength =  string.length() + matchStarts.size() * (replacement.length() - searchLength);
+
+    StringBuilder resultBuilder;
+    resultBuilder.reserveCapacity(resultLength);
+
+    size_t lastMatchEnd = 0;
+    for (auto start : matchStarts) {
+        resultBuilder.append(StringView(string).substring(lastMatchEnd, start - lastMatchEnd));
+        if constexpr (substitutions == StringReplaceSubstitutions::Yes) {
+            StringBuilder replacementBuilder;
+            int ovector[2] = { static_cast<int>(start), static_cast<int>(start + searchLength) };
+            substituteBackreferences(replacementBuilder, replacement, string, ovector, nullptr);
```

Comment:
Any reason not to just pass `resultBuilder` directly here?

Reply:
No, I've fixed it. Thank you